### PR TITLE
[REF] mail: `Message`, review and simplify scss

### PR DESCRIPTION
**Pull request by Brieuc-brd**
Part of the overall v16 SCSS optimization/restyle, task-2704984

task-2819763

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/message/message.scss [hunk 1]
```diff
@@ -53,77 +49,55 @@
...
-
-    > p {
-        margin-bottom: 0;
-    }
-
 }
 
-.o_Message_readMoreLess {
-    display: block;
+.o_Message_prettyBody > p {
```
Comment (Brieuc-brd) [inline addons/mail/static/src/components/message/message.scss:83]: I can't handle the `o_Message_prettyBody` elements.

File: addons/mail/static/src/components/message/message.scss [hunk 2]
```diff
@@ -132,110 +106,25 @@
...
-
-.o_Message_authorName {
-    font-weight: $font-weight-bold;
-}
-
-.o_Message_authorRedirect {
-    cursor: pointer;
-}
-
-.o_Message_content .o_mention {
```
Comment (Brieuc-brd) [inline addons/mail/static/src/components/message/message.scss:187]: It seems that this class is no longer used.

**Comment by Brieuc-brd** [comment]
Hello @stefanorigano 👋 
This PR is ready for review, could you take a look at it? 🙂 
As the Frontend doesn't have the BS5 utility classes yet, I wrote for each concerned property a comment to be as clear as possible. 

What do you think ?

**Comment by Brieuc-brd** [comment]
Hello @alexkuhn @seb-odoo 👋 
This PR is ready for review.
As the Frontend doesn't have the BS5 utility classes yet, I wrote for each concerned property a comment to be as clear as possible.

File: addons/mail/static/src/components/message/message.scss (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/components/message/message.scss:187]: IIRC, `o_mention` class was forged during a message post, so that all "mentions" (`@`) had this classname.
Looks like `o_mail_redirect` and `o_channel_redirect` are now used, and no more `o_mention`:
https://github.com/odoo/odoo/blob/master/addons/mail/static/src/models/composer_view.js#L1166

It's possible that this style rule is applicable for older messages that were posted with old mention.
But I don't think it's that problematic for (very) old messages, as mentions are super relevant now and soon after, but becomes quickly irrelevant with time.

It's a bit sad that we don't have good way to upgrade old messages. It's something we are aware and will make code design decisions to solve this problem once and for all, without hacks here and there.

File: addons/mail/static/src/components/message/message.scss [hunk 3]
```diff
@@ -2,38 +2,34 @@
 // Layout
 // ------------------------------------------------------------------
 
+// TODO : When the migration to BS5 for the backend AND the frontend
+// will be done, we could use BS5 classes to handle the position values
+// (eg. position-absolute top-0 end-0 mt-n3)
 .o_Message_actionListContainer {
     @include o-position-absolute($top: - map-get($spacers, 3), $right: 0);
+    z-index: 10; // Place the element in front of the Composer when they overlap
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message/message.scss:10]: This looks like the only change that seems unrelated to scope of PR.
Is it a fix?

**Review by alexkuhn — COMMENTED** [review]
PR looks quite hard to review, due lots of diff in 2 places that makes seeing immediate translation hard to see.

I think this is good. Just the `z-index: 10;` that looks out-of-scope of this PR.

File: addons/mail/static/src/components/message/message.xml [hunk 4]
```diff
@@ -17,25 +17,33 @@
...
                     'o-starred': messageView.message.isStarred,
                     'mt-3': !messageView.isSquashed and messageView.threadView,
+                    'bg-white': (messageView.message.is_discussion or messageView.message.is_notification) and !messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted) and !messageView.isActive,
+                    'bg-100': (!(messageView.message.is_discussion or messageView.message.is_notification) and !messageView.isActive and !messageView.isSelected) or (messageView.isActive and !messageView.isSelected and (messageView.message.is_discussion or messageView.message.is_notification) and !(messageView.message.isHighlighted or messageView.isHighlighted)),
+                    'bg-200': !(messageView.message.is_discussion or messageView.message.is_notification) and messageView.isActive and !messageView.isSelected,
+                    'bg-400': messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted),
+                    'opacity-50': (messageView.threadView and messageView.threadView.replyingToMessageView) and !messageView.isSelected,
```
[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/mail/static/src/components/message/message.xml:24]: It's ok to keep it that way for now. It's a literal translation with nested SCSS rules. At least there's no more CSS specificity, which is quite an improvement.

I think some concepts are lacking in the modelling, but it's not urgent to do it right now in this PR. Just a comment for myself there's something to improve here in the future :)

File: addons/mail/static/src/components/message/message.scss (hunk 3 above)
Comment (Brieuc-brd) [reply addons/mail/static/src/components/message/message.scss]: This fix has been moved here : https://github.com/odoo/odoo/pull/89610

**Comment by Brieuc-brd** [comment]
Hello @alexkuhn 👋 
This PR is updated 👍

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
